File: app/src/main/java/com/aso/senote/data/source/NotesRepository.java
/*
 * Copyright (c) ASO 2020.
 */

package com.aso.senote.data.source;

import androidx.lifecycle.LiveData;

import com.aso.senote.data.models.Note;
import com.aso.senote.data.models.Notebook;

import java.util.List;

public interface NotesRepository {

    LiveData<List<Notebook>> observeNotebooks();

    LiveData<List<Note>> observeNotes();

    LiveData<List<Note>> observeNotesForNotebook(long notebookId);

    LiveData<Note> observeNoteById(long noteId);

    void saveNote(Note note);

    void saveNotebook(Notebook notebook);

    void deleteNotes(Note... notes);

    void deleteNotebooks(Notebook... notebooks);

    void deleteAllNotes();

    void deleteAllNotebooks();

}

File: app/src/main/java/com/aso/senote/data/source/NotesRepository.kt
/*
 * Copyright (c) ASO 2020.
 */
package com.aso.senote.data.source

import androidx.lifecycle.LiveData
import com.aso.senote.data.models.Note
import com.aso.senote.data.models.Notebook

interface NotesRepository {
    fun observeNotebooks(): LiveData<List<Notebook>?>
    fun observeNotes(): LiveData<List<Note>?>
    fun observeNotesForNotebook(notebookId: Long): LiveData<List<Note>?>
    fun observeNoteById(noteId: Long): LiveData<Note?>
    suspend fun getNote(noteId: Long): Note?
    suspend fun saveNote(note: Note): Long
    suspend fun saveNotebook(notebook: Notebook)
    fun deleteNotes(vararg notes: Note)
    fun deleteNotebooks(vararg notebooks: Notebook)
    fun deleteAllNotes()
    fun deleteAllNotebooks()
}